packages/serverless-contracts/src/contracts/apiGateway/testUtils/mockHandlerInput.ts

```diff
@@ -104,3 +108,14 @@ export const getMockHandlerInput = <
 
 const getFakerDefault = (schema?: Schema) =>
   schema !== undefined ? JSONSchemaFaker.generate(schema) : undefined;
+
+const keepUnchangedRefsOnly = (
+  objValue: unknown,
+  srcValue: unknown,
+): unknown => {
+  if (_.isPlainObject(objValue)) {
+    return _.assignWith({}, objValue, srcValue, keepUnchangedRefsOnly);
+  }
+
+  return undefined;
+};
```

Comment:
we might want to unit test this, I'm not 100% sure what it does

Reply:
:magic_wand:
This is kinda tested by the unit test you commented
But yeah I could split it and do some unit tests